src/beamlime/constructors/inspectors.py

```diff
@@ -37,26 +37,38 @@ def validate_annotation(annotation: Any) -> Literal[True]:
 
 
 Product = TypeVar("Product")
+
+
+def _is_type_newtype_py310(tp: Any) -> bool:
+    from typing import NewType
+
+    return isinstance(NewType, type) and isinstance(tp, NewType)
```

Comment:
What is the first check for?

Reply:
Because `NewType` is a function not a type, in python 3.9 or below.